src/Adapter/Attachment/CommandHandler/CreateAttachmentHandler.php

```diff
@@ -0,0 +1,92 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Attachment\CommandHandler;
+
+use Attachment;
+use PrestaShop\PrestaShop\Core\Domain\Attachment\Command\CreateAttachmentCommand;
+use PrestaShop\PrestaShop\Core\Domain\Attachment\CommandHandler\CreateAttachmentHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Attachment\Exception\AttachmentConstraintException;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+/**
+ * Class CreateAttachmentHandler
+ */
+final class CreateAttachmentHandler implements CreateAttachmentHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @throws AttachmentConstraintException
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    public function handle(CreateAttachmentCommand $command)
+    {
+        $attachment = new Attachment();
+
+        $this->createAttachmentFromCommandData($attachment, $command);
+    }
+
+    /**
+     * @param Attachment $attachment
+     * @param CreateAttachmentCommand $command
+     *
+     * @throws AttachmentConstraintException
+     * @throws \PrestaShopException
+     */
+    private function createAttachmentFromCommandData(Attachment $attachment, CreateAttachmentCommand $command)
+    {
+        if (null !== $command->getUniqueFileName()) {
+            $attachment->file = $command->getUniqueFileName();
+        }
+
+        if ($command->getFile() instanceof UploadedFile) {
+            $attachment->file_name = $command->getFile()->getClientOriginalName();
+        }
+
+        if (null !== $command->getLocalizedDescriptions()) {
+            $attachment->description = $command->getLocalizedDescriptions();
+        }
+
+        if (null !== $command->getLocalizedNames()) {
+            $attachment->name = $command->getLocalizedNames();
+        }
+
+        if (null !== $command->getMimeType()) {
+            $attachment->mime = $command->getMimeType();
+        }
+
+        if (!$attachment->validateFields(false)) {
```

Comment:
I think you also need to call `$attachment->validateFieldsLang(false)` to validate multilingual fields

Reply:
yes although in my opinion, why do we need to call this? When legacy stuff is added it anyway calls this function inside. This validation does not tell which field is not valid. In my opinion, I would add validation in command, and handler level and not use ObjectModel validations at all - it will be easier to refactor later on this part . What do you guys think ? @rokaszygmantas , @sarjon

Reply:
I think it's used to avoid rewriting so much legacy logic yet, because validation includes many steps in legacy object model, such as checking required fields and checking all rules of object model definition